ci-operator/step-registry/upi/install/azure/upi-install-azure-commands.sh

```diff
@@ -196,10 +221,15 @@ az deployment group create -g $RESOURCE_GROUP \
   --parameters baseName="$INFRA_ID"
 
 echo "Deploying 03_infra"
-az deployment group create -g $RESOURCE_GROUP \
-  --template-file "03_infra.json" \
-  --parameters privateDNSZoneName="${CLUSTER_NAME}.${BASE_DOMAIN}" \
-  --parameters baseName="$INFRA_ID"
+cmd="az deployment group create -g ${RESOURCE_GROUP} \
+    --template-file '03_infra.json' \
+    --parameters privateDNSZoneName='${CLUSTER_NAME}.${BASE_DOMAIN}' \
+    --parameters baseName='${INFRA_ID}'"
+if [ -v vnet_basename ] && [ -n "${vnet_basename}" ]; then
+  echo "Using an existing vnet..."
+  cmd="${cmd} --parameters vnetBaseName='${vnet_basename}'"
+fi
```

Comment:
I would like to try to find a better pattern for passing in optional parameters rather than using conditionals.

```
    --parameters -p                               : Supply deployment parameter values.
        Parameters may be supplied from a file using the `@{path}` syntax, a JSON string, or as
        <KEY=VALUE> pairs. Parameters are evaluated in order, so when a value is assigned twice, the
        latter value will be used. It is recommended that you supply your parameters file first, and
        then override selectively using KEY=VALUE syntax.
```
It looks like parameters can accept a JSON string. If passing an empty JSON string does not break the command, perhaps we could create an empty JSON string that is passed to the commands and then conditionally inject the `vnetBaseName` key value pair into the string.

Reply:
That sounds like "conditionally inject the vnetBaseName key value pair into a JSON string" vs "conditionally inject the vnetBaseName parameter for az command line", is there much difference between them, am I understand it right?

Reply:
In this PR, 4 places (03_infra, 04_bootstrap, 05_master, 06_worker) need the optional `vnetBaseName` parameter, if here we just want to reduce conditions for each (or most) commands, we can do that similar things using KEY=VALULE parameters, not have to use a json string parameter. Something like:

```
az_deployment_optional_parameters=""

provisioned_vnet_file="${SHARED_DIR}/customer_vnet_subnets.yaml"
if [ -f "${provisioned_vnet_file}" ]; then
    echo "vnet already created"
    vnet_name=$(python3 -c "import yaml;data = yaml.full_load(open('${provisioned_vnet_file}'));print(data['platform']['azure']['virtualNetwork'])")
    vnet_basename=$(echo "${vnet_name}" | sed 's/-vnet$//')

    az_deployment_optional_parameters="--parameters vnetBaseName='${vnet_basename}'"

    [ -z "${vnet_basename}" ] && echo "Did not get vnet basename" && exit 1
fi
<--snip-->
echo "Deploying 03_infra"
az deployment group create -g $RESOURCE_GROUP \
  --template-file "03_infra.json" \
  --parameters privateDNSZoneName="${CLUSTER_NAME}.${BASE_DOMAIN}" \
  --parameters baseName="$INFRA_ID" ${az_deployment_optional_parameters}
```

That will keep the original parameter KEY=VALUE format, also reduce conditions. WDTY?

Reply:
Yes, this seems like a good approach.

Reply:
Done.